util/eventbus/subscribe.go

```diff
@@ -252,3 +253,49 @@ func (s *Subscriber[T]) Close() {
 	s.stop.Stop() // unblock receivers
 	s.unregister()
 }
+
+// A SubscriberFunc delivers one type of event from a [Client].
+// Events are forwarded synchronously to a function provided at construction.
+type SubscriberFunc[T any] struct {
+	stop       stopFlag
+	read       func(T)
+	unregister func()
+}
+
+func newSubscriberFunc[T any](r *subscribeState, f func(T)) *SubscriberFunc[T] {
+	return &SubscriberFunc[T]{
+		read:       f,
+		unregister: func() { r.deleteSubscriber(reflect.TypeFor[T]()) },
+	}
+}
+
+// Close closes the SubscriberFunc, indicating the caller no longer wishes to
+// receive this event type.  After Close, no further events will be passed to
```

Comment:
Suggestion: reword to make more explicit happens-before statements. Does an in-flight callback finish executing before Close returns? Am I guaranteed that no further callbacks are invoked after the point where Close returns?

I believe/hope the answer is: any in-flight callback returns before Close returns, and no further callbacks will be invoked after Close returns. IOW, Close fully quiesces the SubscriberFunc and the caller is free to clean up associated state without fear.

Reply:
I will leave this alone for now, since it's pre-existing and I think the comment as-written is correct. But it's something we can revisit soon if you feel it's important.